classes/Company.php

```diff
@@ -0,0 +1,224 @@
+            $stmt->execute();
+            $total_candidates = $stmt->fetch(PDO::FETCH_ASSOC);
+            return $total_candidates['total'];
+        }catch(PDOException $e){
+            $timestamp = date('Y-m-d H:i:s'); 
+            error_log("$timestamp: Something went wrong while getting total number of companies from the database! \n", 3, 'error_log');
+        }
+    }
+
+    public function getTotalCompaniesForSearch($search_string) {
+        try {
+            $query = "SELECT COUNT(*) as total FROM companies WHERE 
+                company_name LIKE :search OR 
+                email LIKE :search OR 
+                country LIKE :search OR 
+                city LIKE :search OR 
+                specialization LIKE :search";
+            $stmt = $this->conn->prepare($query);
+            $search_param = "%{$search_string}%";
+            $stmt->bindParam(':search', $search_param);
+            $stmt->execute();
+            $row = $stmt->fetch(PDO::FETCH_ASSOC);
+            return $row['total'];
+        } catch (PDOException $e) {
+            $timestamp = date('Y-m-d H:i:s'); 
+            error_log("$timestamp: Something went wrong while getting total number of companies for search from the database! \n", 3, 'error_log'); 
+        }
+    }
+
+    public function getCompanyDetailsById($id) {
+        
+        try {
+            $stmt = $this->conn->prepare("SELECT * FROM " . $this->table . " WHERE id = :id");
+            $stmt->bindParam(':id', $id, PDO::PARAM_INT);
+            $stmt->execute();
+            
+            $candidate = $stmt->fetch(PDO::FETCH_ASSOC);
+            
+            return $candidate;
+        } catch (PDOException $e) {
+            $timestamp = date('Y-m-d H:i:s'); 
+            error_log("$timestamp: Something went wrong while getting company details by their Id from the database! \n", 3, 'error_log');
+        }
+    }
+
+    public function findCompaniesByMultipleParameters($search_string, $limit = 10, $offset = 0){
+        try{
+
+            $query = "SELECT * FROM companies WHERE 
+                company_name LIKE :search OR 
+                email LIKE :search OR 
+                country LIKE :search OR 
+                city LIKE :search OR 
+                specialization LIKE :search
+                ORDER BY created_at DESC
+                LIMIT :limit OFFSET :offset";
+            $stmt = $this->conn->prepare($query);
+            $search_param = "%{$search_string}%";
+            $stmt->bindParam(':search', $search_param);
+            $stmt->bindParam(':limit', $limit, PDO::PARAM_INT);
+            $stmt->bindParam(':offset', $offset, PDO::PARAM_INT);
+            $stmt->execute();
+            return $stmt;
+            
+        }catch(PDOException $e){
+            $timestamp = date('Y-m-d H:i:s'); 
+            error_log("$timestamp: Something went wrong while searching for company with multiple paramaters in the database! \n", 3, 'error_log'); 
```

Comment:
Log the actual exception $e so we can see the issue in the log file for troubleshooting.